qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,380 @@
+        self,
+        ansatz: QuantumCircuit,
+        operator: BaseOperator | PauliSumOp,
+    ) -> tuple[Callable[[np.ndarray], float | list[float]], dict]:
+        """Returns a function handle to evaluate the energy at given parameters for the ansatz.
+        This is the objective function to be passed to the optimizer that is used for evaluation.
+        Args:
+            ansatz: The ansatz preparing the quantum state.
+            operator: The operator whose energy to evaluate.
+        Returns:
+            Energy of the Hamiltonian of each parameter.
+        Raises:
+            AlgorithmError: If the primitive job to evaluate the energy fails.
+        """
+        num_parameters = ansatz.num_parameters
+
+        # avoid creating an instance variable to remain stateless regarding results
+        eval_count = 0
+
+        def evaluate_energy(parameters):
+            nonlocal eval_count
+
+            # handle broadcasting: ensure parameters is of shape [array, array, ...]
+            parameters = np.reshape(parameters, (-1, num_parameters)).tolist()
+            batchsize = len(parameters)
+
+            try:
+                job = self.estimator.run(batchsize * [ansatz], batchsize * [operator], parameters)
+                values = job.result().values
+            except Exception as exc:
+                raise AlgorithmError("The primitive job to evaluate the energy failed!") from exc
+
+            # TODO recover variance from estimator if has metadata has shots?
+
+            if self.callback is not None:
+                for params, value in zip(parameters, values):
+                    eval_count += 1
+                    self.callback(eval_count, params, value, 0.0)
+
+            return values[0] if len(values) == 1 else values
+
+        return evaluate_energy
+
+    def _get_evalute_gradient(
+        self,
+        ansatz: QuantumCircuit,
+        operator: BaseOperator | PauliSumOp,
+    ) -> tuple[Callable[[np.ndarray], np.ndarray]]:
+        """Get a function handle to evaluate the gradient at given parameters for the ansatz.
+
+        Args:
+            ansatz: The ansatz preparing the quantum state.
+            operator: The operator whose energy to evaluate.
+
+        Returns:
+            A function handle to evaluate the gradient at given parameters for the ansatz.
+
+        Raises:
+            AlgorithmError: If the primitive job to evaluate the gradient fails.
+        """
+
+        def evaluate_gradient(parameters):
+            # broadcasting not required for the estimator gradients
+            try:
+                result = self.gradient.run([ansatz], [operator], [parameters]).result()
+                gradients = result.gradients
+            except Exception as exc:
+                raise AlgorithmError("The primitive job to evaluate the gradient failed!") from exc
+
+            return gradients[0]
+
+        return evaluate_gradient
+
+    def _check_operator_ansatz(
```

Comment:
This method makes no use of `self`. Why is it part of the class whereas e.g. `_validate_initial_point` is not?

Reply:
I've now re-written this to use `self` by moving the default ansatz setting inside `_check_operator_ansatz`. I might move it to a standalone function again in order to make it a common util function alongside [these](https://github.com/Qiskit/qiskit-terra/pull/8640/files#r972980635).